### PR TITLE
Add & remove

### PR DESCRIPTION
**Pull request by elmar8287**
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).

### CODE REVIEW

**Review by abruzy — CHANGES_REQUESTED** [review]
```markdown
Hi @elmar8287 👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_1. Kindly fix your edit, delete and clear all completed functionality as these are not working, only the add functionality works fine._
_2. Kindly add more descriptive instructions on how to get a copy of your app running locally in your readme file._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- implementing keypress for submitting a new todo.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js [hunk 1]
```diff
@@ -161,10 +161,6 @@ checkboxes.forEach((chbox) => {
   });
 });
 
-// function editTask() { // this is function for edit task
```
Comment (Davidosky007) [inline src/index.js:164]: * As the previous reviewer pointed out the edit, clear all completed, delete, completed status are not working and the local storage isn't updated with the status change 
* I get this error in the console when i click on the edit button and the check box kindly 🙏 fix it 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
# Hi @elmar8287 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 1 above)
Comment (elmar8287) [reply src/index.js:164]: Dear @avidosky007,

I fixed the edit delete function, please, check.

My best regards,

Elmar

Comment (elmar8287) [reply src/index.js:164]: image

Comment (elmar8287) [reply src/index.js:164]: As you can see on this screenshot, there is no error when checking the task in the console.

Comment (henatan99) [reply src/index.js:164]: - [x] Dear Elmar, nice work so far. But this issue as pointed out by the previous reviewer is not resolved, kindly see the comments I leave to make the necessary changes.

File: src/index.js [hunk 2]
```diff
@@ -161,10 +161,6 @@ checkboxes.forEach((chbox) => {
...
-//   document.getElementsByClassName('description').contentEditable = true;
-// }
-
 const editTask = index => {
   todoItemElems[index].contentEditable = true;
 };
```
Comment (henatan99) [inline src/index.js:166]: - [x] As you can see here this `editTask` method is not associated with an event listener nor has it a callback in any event listener function. You can notice your `addTask` method is working properly but `editTask`, `deleteTask` and `completeTask` functions are not working. You can see the attached screenshot for the returned error.
 
Screenshot from 2021-08-01 05-42-05

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi Elmar, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (elmar8287) [reply src/index.js:166]: Dear @henatan99,

Are you check8bg the live demo https://elmar8287.github.io/todo-list/  all the mentioned functions are working.

Comment (henatan99) [reply src/index.js:166]: I am checking it. Just give me a minute.

Comment (henatan99) [reply src/index.js:166]: Dear @elmar8287, I have checked the live demo and the `deleteTask` and `completeTask` are working properly. But still, the `editTask` method does not update the local storage as it returns to its initial state when the page is refreshed. In addition, the `Clear all completed` button is not working at all and I don't think you have implemented the method. 
Honestly speaking I don't know why it is not working in the localhost and it is working on github.io but I also noticed those methods ( `editTasks`, ...) are just pure functions and are not associated with an event. You can correct me if I missed something. 
My suggestion, for now, is just to work on the issues which are failing on the live demo also (`editTask` and `clearAllCompleted` methods). After making these changes, kindly make a request for another review. I wish you good luck! :pray:

Comment (elmar8287) [reply src/index.js:166]: Dear @henatan99,

This is my last review ask attempt. Also this is my week repeating.

Comment (elmar8287) [reply src/index.js:166]: Dear @henatan99! 🥇 

Many thanks for the review. I appreciate your comments 👍 

Many thanks and regards,

Elmar

**Pull request merged**